templates/upgrade-job.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{.Job.JobName}}
```

Comment:
Need to add the owner reference to be the CDAP CR that triggers the running of this job.

Reply:
We've set ttlSecondsAfterFinished and ttlSeconds so this job will clean up on its own. According to documentation, Jobs are expected to clean up on their own. https://kubernetes.io/docs/concepts/workloads/controllers/#job

Reply:
It is more than just cleanup. First, having owner reference will get the job get killed immediately on the deletion of the CDAP resource, in which resources are freed much faster. Second, it is about being a good citizen in the cluster to work nicely with other tools (e.g. a visualization tools that shows relations between resources).